Celeste.Mod.mm/Patches/LevelLoader.cs

```diff
@@ -190,8 +203,38 @@ public patch_LevelLoader(Session session, Vector2? startPosition = default)
                 LoadingThread();
             } catch (Exception e) {
                 string sid = session?.Area.GetSID() ?? "NULL";
-                patch_LevelEnter.ErrorMessage = Dialog.Get("postcard_levelloadfailed").Replace("((sid))", sid);
-                Logger.Log(LogLevel.Warn, "LevelLoader", $"Failed loading map {sid}");
+                if (patch_LevelEnter.ErrorMessage == null) {
+                    if (e is KeyNotFoundException && e.StackTrace.Contains("TileHandler") && e.Data.Count == 3) {
+                        int x = (int)e.Data["x"];
+                        int y = (int)e.Data["y"];
+                        char id = (char)e.Data["id"];
+
+                        string room = "";
+                        for (int i = 0; i < GFX.FGAutotiler.LevelBounds.Count; i++) {
+                            if (GFX.FGAutotiler.LevelBounds[i].Contains(x, y)) {
+                                room = session.MapData.Levels[i].Name;
+                                x -= GFX.FGAutotiler.LevelBounds[i].X;
+                                y -= GFX.FGAutotiler.LevelBounds[i].Y;
+                                break;
+                            }
+                        }
+
+                        string type = "";
+                        if (e.StackTrace.Contains("SolidTiles")) {
```

Comment:
What if it's a background tile error, but the stacktrace contains my new mod `SolidTilesStrackTracePolluter`? :3dpausecorn: I think these StackTrace.Contains checks are generally not ideal.